### ISSUE TITLE
no-unused-variable false positive when referenced as a type (rather than a value)

### ISSUE THREAD

**Issue by adidahiya** [palantir/tslint#613]
code samples below, reported in #191 

from @SenHeng:

```ts
import Foo = this.that.Foo;

module some.module.blah {
 export  class bar {
    private foo: Foo;
  }
}
```

`Foo` is a false positive

---

from @nikklassen:

```ts
import DateTimeOpts = Intl.DateTimeFormatOptions;

interface MyDateTimeOpts extends DateTimeOpts {
    timezoneOffset: number;
}

let opts: MyDateTimeOpts;
console.log(opts.timezoneOffset - 1);
```

`DateTimeOpts` is a false positive
